src/views/FormView.vue

```diff
@@ -0,0 +1,381 @@
+import FormInput from "@/components/FormInput.vue";
+
+export default {
+  components: {
+    MDBRadio,
+    MDBInput,
+    MDBBtn,
+    MDBIcon,
+    MDBBtnClose,
+    FormInput,
+  },
+
+  setup() {
+    const cardiomyopathyDataStore = useCardiomyopathyDataStore();
+
+    const genes = ref([]);
+
+    const newGeneName = ref(null);
+    const newGeneComputedName = computed({
+      get: () => newGeneName.value,
+      set: (newName) => {
+        newGeneName.value = newName.toUpperCase();
+      },
+    });
+
+    function addGene() {
+      if (newGeneName.value) {
+        genes.value.push(newGeneName.value);
+        newGeneName.value = null;
+      }
+    }
+
+    const form = reactive({
+      cardiomyopathyType: "",
+      ledv: "",
+      redv: "",
+      lesv: "",
+      resv: "",
+      lvef: "",
+      rvef: "",
+      lvmass: "",
+      rvmass: "",
+      lsv: "",
+      rsv: "",
+      scar: "",
+      gender: "",
+      mriAge: "",
+      apicalHCM: "",
+      suddenCardiacDeath: "",
+      hypertension: "",
+      diabetes: "",
+      myectomy: "",
+    });
+    const submitData = async () => {
+      genes.value.forEach((gene) => {
+        cardiomyopathyDataStore
+          .addData(form.cardiomyopathyType, gene, form)
+          .then(() => {})
+          .catch((error) => {
+            /*Set error variable using error.message*/
+          });
+      });
+      await addData({ ...form });
+      form.cardiomayopathyType = "";
+      form.ledv = "";
+      form.redv = "";
+      form.lesv = "";
+      form.resv = "";
+      form.lvef = "";
+      form.rvef = "";
+      form.lvmass = "";
+      form.rvmass = "";
+      form.lsv = "";
+      form.rsv = "";
+      form.scar = "";
+      form.gender = "";
+      form.ageAtMRI = "";
+      form.apicalHCM = "";
+      form.suddenCardiacDeath = "";
+      form.hypertension = "";
+      form.diabetes = "";
+      form.myectomy = "";
+    };
+
+    return {
+      form,
+      submitData,
+      addGene,
+      newGeneComputedName,
+      genes,
+      cardiomyopathyTypes: cardiomyopathyDataStore.types,
+    };
+  },
+};
+</script>
+
+<template>
+  <main>
+    <h1>Cardiomyopathy Form</h1>
+    <div class="delete-data-nav">
+      <MDBBtn color="danger">Delete data</MDBBtn>
+    </div>
+    <div class="add-data">
+      <form @submit.prevent="submitData">
+        <select
+          class="form-select"
+          v-model="form.cardiomyopathyType"
+          aria-label="Cardiomyopathy type select field"
+          required
+        >
+          <option value disabled>Cardiomyopathy type</option>
+          <option
+            v-for="(text, type) in cardiomyopathyTypes"
+            :key="type"
+            :value="type"
+          >
+            {{ text }}
+          </option>
+        </select>
+
+        <MDBInput
+          label="Left ventricular end diastolic volume"
+          v-model.number="form.ledv"
+          type="number"
+          min="0"
+          required
+        />
```

Comment:
Switch structure to match rest of site. You will need to import some new components and define the formChecked variable to be set to true on validation submission if validation fails.
```suggestion
  <main id="add-data-view">
    <h1 class="text-center">Add data</h1>
    <MDBRow
      tag="form"
      class="g-3 mb-3 needs-validation"
      :class="formChecked && 'was-validated'"
      novalidate
      @submit.prevent="submitData"
    >
      <hr class="mb-0" />
      <MDBCol md="12">
        <select
          class="form-select"
          v-model="form.cardiomyopathyType"
          aria-label="Cardiomyopathy type select field"
          required
        >
          <option value disabled>Cardiomyopathy type</option>
          <option
            v-for="(text, type) in cardiomyopathyTypes"
            :key="type"
            :value="type"
          >
            {{ text }}
          </option>
        </select>
      </MDBCol>
      
      <FormInput
        v-model.number="form.ledv"
        label="Left ventricular end diastolic volume"
        md="6"
        :formChecked="formChecked"
        required
      />
...
```